### DISCUSSION TITLE
Add directories recursively

### DISCUSSION THREAD

**Discussion by Siggless** [MoeMusic/Moe#277] [Ideas]
As far as I can tell there is no way to add multiple albums at once.
This makes adding lots a new music, especially initially adding an existing library, laborious.

Could there be an option to add multiple albums found in all directories under a given path?
I have used this code which seems to work in my case, however I have had to edit this into the `add_cli.py`, and this assumes that the music is organised in album directories:
```python
def _add_path_recursive(session: Session, path: Path):
    """Adds multiple items to the library from a given path.
        This looks recursively through sub-directories, adding all files to an
        album of the containing directory's name.
    
    Args:
        session: Library db session.
        path: Path to add. Must be a directory and not a file.

    Raises:
        AddError: Path not found or other issue adding the item to the library.
        AlbumError: Could not create an album from the given directory.
        SkipAdd: External program or user elected to skip adding the item.
    """
    if path.is_file():
        raise AddError(
            f"Adding recursively requires a directory, not a file. [{path=}]"
        ) from None
    
    # Walk through all sub-directories and if there are files, make an album
    #  from all files in the single directory.
    for (dirpath, dirnames, filenames) in Path.walk(path):
        log.info(f"Searching directory {str(dirpath)}")
        if len(filenames) > 0: # Should also make sure not our library_path
            try:
                moe.add.add_item(session, Album.from_dir(dirpath))
            except AlbumError:
                log.error(f"No files found in non-empty directory {dirpath}")
```

**Comment by jtpavlock** [comment]
You can add any number of albums or tracks at once using the [moe add](https://mrmoe.readthedocs.io/en/latest/cli.html#add) command, it just doesn't understand nested sub-directories of albums. I'm not sure what shell you're using, but there's typically commands that can list out nested directories, which can then be passed to moe.

I didn't want to directly give the option to recursively add directories as albums, as many albums include directories for attachments like cover art, or for multiple discs, which complicates trying to discern if a directory is an album or not.
